cmd/kyma/sync/function/opts.go

```diff
@@ -0,0 +1,50 @@
+package function
+
+import (
+	"fmt"
+	"github.com/kyma-project/cli/internal/cli"
+	"os"
+)
+
+//Options defines available options for the command
+type Options struct {
+	*cli.Options
+
+	Name       string
+	Namespace  string
+	OutputPath string
+}
+
+//NewOptions creates options with default values
+func NewOptions(o *cli.Options) *Options {
+	options := &Options{Options: o}
+	return options
+}
+
+func (o *Options) setDefaults(defaultNamespace string) (err error) {
+	if o.OutputPath == "" {
+		o.OutputPath, err = os.Getwd()
+		if err != nil {
+			return err
+		}
+	}
+
+	setIfZero(&o.Namespace, defaultNamespace, "default")
+
+	if o.Name == "" {
+		return fmt.Errorf("flag 'name' is required")
```

Comment:
I think you can use simple `MarkFlagRequired` instead of this `if`
https://pkg.go.dev/github.com/spf13/cobra#Command.MarkFlagRequired

Reply:
Good point, however there is a catch with `MarkFlagRequired`function, it returns `error`.
Since this method needs to be invoked in [NewCmd](https://github.com/kyma-project/cli/pull/618/files#diff-68b06847f40033401830cc910c8ccea68ddbf92fe2991aa7044195b3540bc16cR21) instead in [Run](https://github.com/kyma-project/cli/pull/618/files#diff-68b06847f40033401830cc910c8ccea68ddbf92fe2991aa7044195b3540bc16cR43) we would have to: 
- _either_ `panic` in case of an error 
- _or_ change the signature of the `NewCmd` method to return an `error` and this leads to the refactor of all the code that starts in [main](https://github.com/kyma-project/cli/blob/master/cmd/main.go#L17) function of the CLI.

I believe the 2nd option is better but it is out of the scope of this PR, let's address it in separate issue.